Mathlib/Tactic/Simps/Basic.lean

```diff
@@ -474,6 +474,25 @@ def findProjectionIndices (strName projName : Name) : MetaM (List Nat) := do
   let allProjs := pathToField ++ [fullProjName]
   return allProjs.map (env.getProjectionFnInfo? · |>.get!.i)
 
+/-- We do not consider `toFoo` to be a prefix to `toFoo_1`, as the latter is often
+a different field with an auto-generated name. -/
+private def dropPrefixIfNotNumber? (s : String) (pre : Substring) : Option Substring := do
+  let ret ← Substring.dropPrefix? s pre
+  ret.toString.data.head?.elim ret
+    (fun x ↦ if 48 ≤ x.val ∧ x.val < 58 then none else return ret)
+
+private def isPrefixOfAndNotNumber (s p : String) : Bool := (dropPrefixIfNotNumber? p s).isSome
+
+private def splitOnNotNumber (s delim : String) : List String :=
+  (process (s.splitOn delim).reverse "").reverse where
+    process (arr : List String) (tail : String) := match arr with
+      | [] => []
+      | (x :: xs) =>
+        if (x.data.head?.elim false (fun x ↦ 48 ≤ x.val ∧ x.val < 58)) then
+          process xs (tail ++ delim ++ x)
+        else
+          List.cons (x ++ tail) (process xs "")
```

Comment:
This function is actually not needed to make the test work. I don't really understand the simps code that much to propose a more precise test that could test this.

Reply:
I just realized that previous occasions of `String.splitOn s "_"` were all for tracing purposes and that's why I failed to test them. The tests are sill useful though and I'll add them. Thanks!